src/server/dflycmd.cc

```diff
@@ -651,6 +658,37 @@ void DflyCmd::StopReplication(uint32_t sync_id) {
   replica_infos_.erase(sync_id);
 }
 
+void DflyCmd::BreakStalledFlowsInShard() {
+  unique_lock lk(mu_, try_to_lock);
+  if (!lk.owns_lock())
```

Comment:
best effort. Each flow will be probed periodically. With high probabilty (unless we have a deadlock) this lock will be grabbed.

Reply:
This should be a comment, instead of the short `// give up` below

Reply:
Take into account that you might be holding the global lock for a dangerously long time, you can copy the shared pointers into a separate vector and then iterate over it

Reply:
Let's see if it's a problem at all